s3/src/main/java/io/ddf/s3/S3DDF.java

```diff
@@ -0,0 +1,135 @@
+package io.ddf.s3;
+
+import io.ddf.DDF;
+import io.ddf.DDFManager;
+import io.ddf.content.Schema;
+import io.ddf.content.SqlResult;
+import io.ddf.datasource.DataFormat;
+import io.ddf.exception.DDFException;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Created by jing on 12/2/15.
+ */
+public class S3DDF extends DDF {
+    // Whether this ddf has header.
+    private Boolean mHasHeader = false;
+    // It's a directory or file.
+    private Boolean mIsDir;
+    // The format of this s3ddf. If it's a folder, we requires that all the files in the folder should have the same
+    // format, otherwise the dataformat will be set to the dataformat of the first file under this folder.
+    private DataFormat mDataFormat;
+    // Schema String.
+    private String mSchemaString;
```

Comment:
Should this be a string or is it more useful to parse it into a Schema object up front.